papyrus_ogcproxy/views.py

```diff
@@ -29,37 +37,44 @@
 #
 proxy_info = None
 
+
 def ogcproxy(request):
     url = request.params.get("url")
     if url is None:
-        return HTTPBadRequest()
+        return HTTPBadRequest("Missing url in the query string")
 
     # check for full url
     parsed_url = urlparse(url)
     if not parsed_url.netloc or parsed_url.scheme not in ("http", "https"):
-        return HTTPBadRequest()
+        return HTTPBadRequest("Wrong scheme")
 
     # forward request to target (without Host Header)
     http = Http(
         disable_ssl_certificate_validation=True,
         proxy_info=proxy_info)
-    h = dict(request.headers)
-    h.pop("Host", h)
+
+    headers = {
+        h: request.headers[h]
+        for h in forwarded_headers
+        if h in request.headers
+    }
```

Comment:
I think you can use fewer lines. I am picky but this formatting looks very unusual to me.